src/wms/src/test/java/org/geoserver/wms/legendgraphic/FeatureCountLegendGraphicTest.java

```diff
@@ -162,6 +162,110 @@ public void testStatesFull() throws Exception {
+    public void testStatesMissingHeightWidth() throws Exception {
+        runGetLegendGraphics(
+                "wms?service=WMS&version=1.1.1&request=GetLegendGraphic&format=image/png"
+                        + "&layer=" + getLayerId(SF_STATES)
+                        + "&style=Population"
+                        + "&srs=EPSG:4326&bbox=" + "-130,24,-66,50"
+                        + "&legend_options="
+                        + GetLegendGraphicRequest.COUNT_MATCHED_KEY + ":true");
+        assertEquals(1, ruleSets.size());
+        Rule[] rules = ruleSets.get(0);
+        logLabels(rules);
+        assertEquals(4, rules.length);
+        assertLabel("2M - 4M (10)", rules[0]);
+        assertLabel("< 2M (16)", rules[1]);
+        assertLabel("> 4M (23)", rules[2]);
+        // this is the rule for outline and text symbolizer, Alaska and Hawaii are not in the
+        // map but Washington DC is and it's not a state (50 - 2 + 1)
+        assertLabel("(49)", rules[3]);
+    }
+
+    @Test
+    public void testStatesMissingHeightWidthSrs() throws Exception {
+        runGetLegendGraphics(
+                "wms?service=WMS&version=1.1.1&request=GetLegendGraphic&format=image/png"
+                        + "&layer=" + getLayerId(SF_STATES)
+                        + "&style=Population"
+                        + "&bbox=" + "-130,24,-66,50"
+                        + "&legend_options="
+                        + GetLegendGraphicRequest.COUNT_MATCHED_KEY + ":true");
+        assertEquals(1, ruleSets.size());
+        Rule[] rules = ruleSets.get(0);
+        logLabels(rules);
+        assertEquals(4, rules.length);
+        assertLabel("2M - 4M (10)", rules[0]);
+        assertLabel("< 2M (16)", rules[1]);
+        assertLabel("> 4M (23)", rules[2]);
+        // this is the rule for outline and text symbolizer, Alaska and Hawaii are not in the
+        // map but Washington DC is and it's not a state (50 - 2 + 1)
+        assertLabel("(49)", rules[3]);
+    }
+
+    @Test
+    public void testStatesMissingBboxSrs() throws Exception {
+        runGetLegendGraphics(
+                "wms?service=WMS&version=1.1.1&request=GetLegendGraphic&format=image/png"
+                        + "&layer=" + getLayerId(SF_STATES)
+                        + "&style=Population"
+                        + "&width=550&height=250"
+                        + "&legend_options="
+                        + GetLegendGraphicRequest.COUNT_MATCHED_KEY + ":true");
+        assertEquals(1, ruleSets.size());
+        Rule[] rules = ruleSets.get(0);
+        logLabels(rules);
+        assertEquals(4, rules.length);
+        assertLabel("2M - 4M (10)", rules[0]);
+        assertLabel("< 2M (16)", rules[1]);
+        assertLabel("> 4M (23)", rules[2]);
+        // this is the rule for outline and text symbolizer, Alaska and Hawaii are not in the
+        // map but Washington DC is and it's not a state (50 - 2 + 1)
+        assertLabel("(49)", rules[3]);
+    }
+
+    /*@Test
```

Comment:
Ah, this is due to map generalization, some features become so small that they are not going to be painted. You can fix this by adding 
renderer.setGeneralizationDistance(0); 
in FeatureCountProcessor at line 112 (right after the super.onBeforeRender(renderer);